src/Detector/Go/GoConstants.cs

```diff
@@ -0,0 +1,17 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+namespace Microsoft.Oryx.Detector.Go
+{
+    internal static class GoConstants
+    {
+        internal const string PlatformName = "go";
+        internal const string GoDotModFileName = "go.mod";
```

Comment:
nit:GolangModFileName

Reply:
I do prefer keeping the name as the filename. GolangModFileName. I searched to online for GolandModFileName and did not appear to be a popular naming convention